src/pybind/mgr/cephadm/schedule.py

```diff
@@ -385,6 +385,8 @@ def expand_candidates(ls: List[DaemonPlacement], num: int) -> List[DaemonPlaceme
 
     def find_ip_on_host(self, hostname: str, subnets: List[str]) -> Optional[str]:
         for subnet in subnets:
+            # to normalize subnet
+            subnet = str(ipaddress.ip_network(subnet))
```

Comment:
out of curiosity, what did you see that prompted this change? An IPv6 issue?

Reply:
It was not matching network if passed ipv6 network is not normalised. I was passing network  as fd12:3456:789a:0000::/64 and it was matching it to fd12:3456:789a::/64.